CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -444,6 +469,25 @@ class CameraFragment : Fragment() {
         }
     }
 
+    /** Enabled or disabled a button to switch cameras depending on the available cameras */
+    private fun updateCameraSwitchButton() {
+        val switchCamerasButton = container.findViewById<ImageButton>(R.id.camera_switch_button)
+        try {
+            switchCamerasButton.isEnabled = hasBackCamera() && hasFrontCamera()
+        } catch (exception: CameraInfoUnavailableException) {
```

Comment:
is the `hasCamera` method throwing this exception? What's the difference between the method returning `false` and throwing an exception?

Reply:
`hasCamera()` throws that exception when CameraX can't access cameras, this can happen if the camera permission is revoked for instance. Instead of crashing, I return false.